src/panels/lovelace/components/hui-yaml-editor.ts

```diff
@@ -0,0 +1,98 @@
+// @ts-ignore
+import CodeMirror from "codemirror";
+import "codemirror/mode/yaml/yaml";
+// @ts-ignore
+import codeMirrorCSS from "codemirror/lib/codemirror.css";
+import { fireEvent } from "../../../common/dom/fire_event";
+declare global {
+  interface HASSDomEvents {
+    "yaml-changed": {
+      value: string;
+    };
+    "yaml-save": {};
+  }
+}
+
+export class HuiYamlEditor extends HTMLElement {
+  public codemirror: CodeMirror;
+  private _value: string;
+
+  public constructor() {
+    super();
+    CodeMirror.commands.save = () => {
```

Comment:
This is changing a singleton variable, this means that we can't have 2 editors open at the same time, or even have 1 editor show up in a modal while the other editor is open in the background.